Adds custom error logging

colestrode opened the pull request:
Thanks for this library! We've added it to a few repos and it's been really helpful!

One feature that would be nice for our team would be the ability to add a custom logger. We have some standard logging formats that capture the timestamp and other information. It would be great if this handler would give us a way to use our own format.

I made this PR to add a custom logger to the library and updated the README with the usage. Let me know what you think!

Thanks again!

mcollina commented on make-promises-safe.js:
```diff
@@ -3,11 +3,13 @@
 
 process.on(event, function (err) {
-  console.error(err)
+  module.exports.logger.error(err)
```
I would prefer us to just wrap this as a pure function, not as a `logger` object.

colestrode replied:
so something like this?
```
module.exports.logError = function (err) { console.log(err); }
```

mcollina replied:
I would go with `module.exports.logError = console.error`.

colestrode commented:
I'm having a little trouble fitting a test into this existing framework. Normally I would use some sinon stubs to inspect things like this, would that be alright?

mcollina commented:
you can use [proxyquire](http://npm.im/proxyquire)

colestrode commented:
Not sure proxyquire would help in this case? there's no dependencies. Or maybe I'm missing something?

davidmarkclements commented:
1. make a test fixture file  that exports a logError function 
2. copy paste the current test
3. require.resolve to the fixture (same approach as with main)
4. spawn fixture instead of main 
5. check that log output corresponds to the logError function output

colestrode commented:
I think I got it! Thanks for your help :) I added a test and have it passing. I also added the `--fix` option to standard, this made it a lot easier for me to get things into shape and contribute.

mcollina commented on package.json:
```diff
@@ -4,7 +4,7 @@
   "main": "make-promises-safe.js",
   "scripts": {
-    "test": "standard | snazzy && tap test.js"
+    "test": "standard --fix | snazzy && tap test.js"
```
Can you revert this? If you want, please put it in a separate script. We run `test` before any commit and CI.

colestrode replied:
reverted

mcollina commented on test.js:
```diff
@@ -23,3 +23,16 @@ test('crashes the process on unhandled rejection', (t) => {
+test('logs error', (t) => {
+  t.plan(1)
+
+  const child = spawn(process.execPath, ['-r', main, '-e', 'require("./make-promises-safe").logError = console.log; Promise.reject(new Error(\'kaboom\'))'])
```
Can you customize the `logError` function somehow, maybe prefixing a string?
